protoc-gen-gogo/generator/generator.go

```diff
@@ -549,6 +548,19 @@ var uniquePackageName = make(map[*descriptor.FileDescriptorProto]string)
 // value is the name that appears in the generated code.
 var pkgNamesInUse = make(map[string][]*FileDescriptor)
 
+func pkgNameInUse(pkg string, f *FileDescriptor) bool {
```

Comment:
it's not cleanup - i had to add some logic to see here. This is actually a reduction in the diff with golang/protobuf - take a look at the implementation of `RegisterUniquePackageName` there.